### PR TITLE
fix(Run Info): Allow teachers to request run info by id

### LINKED ISSUE #229: bug(Run Info): Allow teachers to request run info by id
Right now only students can make the request to "/api/student/run/info-by-id". Teachers should also be able to make this request but the url should probably be changed to "/api/user/run/info-by-id" so that it's not student specific.

### PR DESCRIPTION
**Pull request by geoffreykwan**
## Changes

- Moved getRunInfoById() from StudentAPIController.java to its parent UserAPIController.java so that teachers can use it too
- In getRunInfoById(), check that the signed in user is associated with the run before returning the run info
- Moved isUserAssociatedWithRun() to UserServiceImpl.java to avoid duplication

## Test

- Test with https://github.com/WISE-Community/WISE-Client/pull/1327

Closes #229

### CODE REVIEW

File: src/main/java/org/wise/portal/service/user/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -328,4 +329,14 @@ public List<User> retrieveStudentUsersWhoLoggedInThisMonth() {
...
     return userDao.retrieveStudentUsersWhoLoggedInThisYear();
   }
+
+  public boolean isUserAssociatedWithRun(User user, Run run) throws ObjectNotFoundException {
+    if (user.isStudent() && run.isStudentAssociatedToThisRun(user)) {
+      return true;
+    } else if (user.isTeacher() && run.isTeacherAssociatedToThisRun(user)) {
+      return true;
+    }
+    return false;
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/service/user/impl/UserServiceImpl.java]: Change to use one return statement.

**Review by hirokiterashima — CHANGES_REQUESTED** [review]
Functionality works fine. 

See inline comment.

Also, should the code to handle run info requests be in a new RunInfoController, with the endpoint /api/run/{runId}/info?

**Comment by geoffreykwan** [comment]
Yea maybe we should look into making a RunAPIController or RunInfoController in another issue. There are other run related endpoints in StudentAPIController, TeacherAPIController, and UserAPIController. For example /api/student/run/info which shares code with /api/user/run/info-by-id.

**Review by hirokiterashima — APPROVED** [review]
LGTM

**Pull request merged**